### ISSUE TITLE
Training is very slow for MAML and MAML++?

### ISSUE THREAD

**Issue by aniruddhraghu** [AntreasAntoniou/HowToTrainYourMAMLPytorch#5]
Hi,

Firstly, thanks a lot for open-sourcing the code -- great to have a resource for this!

I seem to be having an issue regarding training time for these models. I'm training a 5-way 1-shot system on Omniglot using the supplied code. When comparing with the original MAML implementation for this task (https://github.com/cbfinn/maml), I find that this code trains a lot slower (validation accuracy takes many more epochs and wall-clock time before it reaches a comparable level). Would you know why this might be? I have also found that training a MAML++ system on this task (using the code/config provided) is very slow.

For reference, both setups were run on the same machine with a GPU, and I changed a couple of parameters in the configuration (learning rate and batch size) for the MAML run to match what was used in the original paper.  

Thanks for the help!

**Comment by AntreasAntoniou** [comment]
Hello there,

A bug was introduced 3 days ago, which made things converge much slower than usual. Can you retry the experiments with the current state of the code? Thank you.

Regards, Antreas

**Comment by aniruddhraghu** [comment]
Thanks -- I will try again and let you know.

**Comment by aniruddhraghu** [comment]
I think it now trains faster (in terms of getting better performance with fewer train steps) but it still runs slowly (that is, each training step takes longer than it did with the the original MAML code). Is this something you also experienced?

**Comment by aniruddhraghu** [comment]
I'm on Ubuntu 16.04, using an NVIDIA GeForce GTX 1060 graphics card. I am also using CUDNN, yes.

I'm getting something like 1.2 seconds/iteration using your code, and I think more like 0.4 seconds/iteration using the original MAML code. Turning off logging with tqdm gets some speedup, but it's not very significant.

**Comment by aniruddhraghu** [comment]
I did some more analysis and I'm also observing this issue on a GPU server with a Titan X graphics card; this is when comparing this implementation of MAML with the original.

**Comment by AntreasAntoniou** [comment]
Are you using the same task batch sizes for both? I will investigate this myself. I just don't have the time right now. I can tell you, however, that the speed of the two systems in late June was faster in my implementation. I need time to investigate so let's leave this open until then. Are you using an SSD disk?

**Comment by aniruddhraghu** [comment]
I am, yes -- thanks for taking a look! I am also using an SSD disk.

**Comment by aniruddhraghu** [comment]
I think it's sorted -- will close the issue. Thanks for the help!
